### ISSUE TITLE
refactor Transaction.error field to expected_error

### ISSUE THREAD

**Issue by winsvega** [ethereum/execution-specs#1492]
in Transactoin class this field

error: List[TransactionException] | TransactionException | None = Field(None, exclude=True)

seems to be "expected_error "  the error that we define in tests. need refactor.

I see transaction exception verification. And it looks like tx.error is the error we define ourselves in .py files.

But the actual error reported by the tool is also called error and its confusing

**Comment by felix314159** [comment]
I agree that it's confusing and we [still have not changed it](https://github.com/ethereum/execution-spec-tests/blob/61f8ac90841770d9fc407f1498ec0e5229b27508/src/ethereum_test_types/transaction_types.py#L273)
